core/src/main/java/com/sekwah/advancedportals/core/tags/TriggerBlockTag.java

```diff
@@ -48,4 +55,29 @@ public List<String> autoComplete(String argData) {
 
         return triggerBlocks;
     }
+
+    @Override
+    public boolean created(TagTarget target, PlayerContainer player, String[] argData) {
+        for (int i = 0; i < argData.length; i++) {
+            String material = argData[i].toUpperCase(Locale.ROOT);
+            if (!isValidMaterial(material)) {
+                infoLogger.warning(Lang.translate("tag.triggerblock.error.invalid_material"));
+                return false;
+            }
+            argData[i] = material; // Save the uppercase material name
+        }
+        return true;
+    }
+
+    @Override
+    public void destroyed(TagTarget target, PlayerContainer player, String[] argData) {
+
+    }
+
+    private boolean isValidMaterial(String material) {
+        // Assuming serverContainer has a method to get all valid materials
```

Comment:
Comment not needed as the fact its part of the implementation interface assumes it would be implemented.